include/ddsrouter/event/SignalHandler.hpp

```diff
@@ -129,7 +136,22 @@ class SignalHandler : public EventHandler<int>
     static std::vector<SignalHandler*> active_handlers_;
 
     //! Guards access to variable \c active_handlers_
-    static std::mutex active_handlers_mutex_;
+    static std::recursive_mutex active_handlers_mutex_;
```

Comment:
Why is recursive needed?

Reply:
The mutex is taken in the destructor so it waits for any possible signal handling process to end. It's been made recursive since ``unset_callback```(called from the constructor) also takes this mutex.